exdir/core/exdir_file.py

```diff
@@ -138,10 +161,25 @@ def __init__(self, directory, mode=None, allow_remove=False,
     def close(self):
         """
         Closes the File object.
-        Currently, this has no effect because all data is immediately written to disk.
+        Sets the OpenMode to FILE_CLOSED which denies access to any attribute or
+        child
         """
+        import gc
         # yeah right, as if we would create a real file format
-        pass
+        for name, data_set in self._open_datasets.items():
+            # there are no way to close the memmap other than deleting all
+            # references to it, thus
+            data_set.flush()
+            data_set.setflags(write=False) # TODO does not work
+        self._open_datasets = {}
+        gc.collect()
```

Comment:
Could be good to add a comment here explaining why we need to do gc.collect. It could be easy to remove this by accident later because someone does not know why it is here.